src/templates/events/eventDirectionsSection.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import querystring from 'querystring'
+
+import { Container, Row } from '../../components/grid'
+import { media } from '../../theme/media'
+import theme from '../../theme/theme'
+
+if (
+  process.env.NODE_ENV === 'development' &&
+  !process.env.GATSBY_GOOGLE_MAPS_API_KEY
+) {
+  console.warn(
+    '⚠️ You are missing the GATSBY_GOOGLE_MAPS_API_KEY environment variable'
+  )
+}
+
+const StyledContainer = styled(Container)`
+  padding: 0px 0px 30px;
+  background-color: white;
+  ${media.desktop`
+    padding: 60px 0px;
+    background-color: ${theme.colors.indigo};
+  `};
+`
+
+const Heading = styled.h2`
+  font-size: 1.125rem;
+  font-weight: 300;
+  margin: 0px 10px 15px 10px;
+  color: ${props => props.theme.colors.indigo};
+  ${media.desktop`
+    margin: 0px 15px 15px 15px;
+    font-size: 1.5rem;
+    font-weight: 600;
+    margin-bottom: 30px;
+    color: white;
+  `};
+`
+
+const MapLink = styled.a`
+  display: block;
+  height: 215px;
+  background-size: cover;
+  background-position: center center;
+  margin: 0px;
+  ${media.tablet`
+    height: 350px;
+    margin: 0px 50px;
+  `};
+  ${media.desktop`
+    height: 450px;
+    max-width: 830px;
+    margin: 0px 90px;
+  `};
+`
+
+export class EventDirectionSection extends React.Component {
+  state = { width: null, height: null }
+
+  componentDidMount() {
+    this.componentDidUpdate()
```

Comment:
calling this.componentDidUpdate() seems like a strange pattern?

can you use `static getDerivedStateFromProps` instead? it calls on initial mount and on rerender, and all you need to do is return an object, which updates the state

https://medium.com/@baphemot/whats-new-in-react-16-3-d2c9b7b6193b

Reply:
getDerivedStateFromProps is not applicable here because it is called before a mount, but you're right that it's weird to call componentDidUpdate within componentDidMount. I'll move the sizing logic to another method.